src/Database/DriverInterface.php

```diff
@@ -0,0 +1,260 @@
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.6.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database;
+
+use Cake\Database\Query;
+use Cake\Database\Statement\PDOStatement;
+use InvalidArgumentException;
+use PDO;
+use PDOException;
+
+/**
+ * Interface for database driver.
+ */
+interface DriverInterface
+{
+    /**
+     * Establishes a connection to the database server.
+     *
+     * @return bool True on success, false on failure.
+     */
+    public function connect();
+
+    /**
+     * Disconnects from database server.
+     *
+     * @return void
+     */
+    public function disconnect();
+
+    /**
+     * Returns correct connection resource or object that is internally used.
+     *
+     * @return mixed Connection object used internally.
+     */
+    public function getConnection();
+
+    /**
+     * Set the internal connection object.
+     *
+     * @param mixed $connection The connection instance.
+     * @return $this
+     */
+    public function setConnection($connection);
+
+    /**
+     * Returns whether php is able to use this driver for connecting to database.
+     *
+     * @return bool True if it is valid to use this driver.
+     */
+    public function enabled();
+
+    /**
+     * Prepares a sql statement to be executed.
+     *
+     * @param string|\Cake\Database\Query $query The query to turn into a prepared statement.
+     * @return \Cake\Database\StatementInterface
+     */
+    public function prepare($query);
+
+    /**
+     * Starts a transaction.
+     *
+     * @return bool True on success, false otherwise.
+     */
+    public function beginTransaction();
+
+    /**
+     * Commits a transaction.
+     *
+     * @return bool True on success, false otherwise.
+     */
+    public function commitTransaction();
+
+    /**
+     * Rollbacks a transaction.
+     *
+     * @return bool True on success, false otherwise.
+     */
+    public function rollbackTransaction();
+
+    /**
+     * Get the SQL for releasing a save point.
+     *
+     * @param string $name The table name.
+     * @return string
+     */
+    public function releaseSavePointSQL($name);
+
+    /**
+     * Get the SQL for creating a save point.
+     *
+     * @param string $name The table name.
+     * @return string
+     */
+    public function savePointSQL($name);
+
+    /**
+     * Get the SQL for rollingback a save point.
+     *
+     * @param string $name The table name.
+     * @return string
+     */
+    public function rollbackSavePointSQL($name);
+
+    /**
+     * Get the SQL for disabling foreign keys.
+     *
+     * @return string
+     */
+    public function disableForeignKeySQL();
```

Comment:
Should those be Sql instead of acronym uppercase SQL etc as per our convention everywhere?
It is BC either way, as casing is not sensitive.

PS: disable sounds like void, should this be `get...`? Same above for save, enable etc.